### PR TITLE
Fix for targeting.props

### PR DESCRIPTION
**Pull request by ig-sinicyn**
@NN---  I've updated codegen for targeting.props.
I do not update `<TargetFrameworks>` property of real projects, however

Feel free to change target framework lists in Targeting tests as you wish.

P.S.  If any questions you can contain me via igor.sinicyn@gmail.com or skype igor.sinicyn@outlook.com. I think it will be much faster than chatting in the PR:)

### CODE REVIEW

File: Build/Props/CodeJam.Targeting.props [hunk 1]
```diff
@@ -1,25 +1,30 @@
...
-	<PropertyGroup Condition=" '$(TargetFramework)' == 'net461' ">
+	<!-- We DO most development and testing on netcoreapp3.0 target -->
+	<PropertyGroup Condition=" '$(TargetFramework)' == 'netcoreapp3.0' ">
 		<DefineConstants>$(DefineConstants);DEFAULT_PLATFORM</DefineConstants>
 	</PropertyGroup>
 
-	<!-- Generated by CodeJam.TargetingTests.EmitTargetingConstants -->
+	<!-- Templates for <TargetFrameworks/> project property -->
+	<PropertyGroup>
+		<CopyMeTargetFrameworks>netcoreapp3.0;net5.0;netcoreapp2.0;netcoreapp1.0;netstandard2.1;netstandard2.0;netstandard1.5;netstandard1.3;net472;net461;net45;net40;net35;net20</CopyMeTargetFrameworks>
```
Comment (NN---) [inline Build/Props/CodeJam.Targeting.props:22]:
```suggestion
		<CopyMeTargetFrameworks>net5.0;netcoreapp3.0;netcoreapp2.0;netcoreapp1.0;netstandard2.1;netstandard2.0;netstandard1.5;netstandard1.3;net472;net461;net45;net40;net35;net20</CopyMeTargetFrameworks>
```

File: Build/Props/CodeJam.Targeting.props [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
+		<CopyMeTestTargetFrameworks>netcoreapp3.0;net5.0;netcoreapp2.1;netcoreapp1.1;net472;net461;net45;net40;net35;net20</CopyMeTestTargetFrameworks>
```
Comment (NN---) [inline Build/Props/CodeJam.Targeting.props:23]:
```suggestion
		<CopyMeTestTargetFrameworks>net5.0;netcoreapp3.0;netcoreapp2.1;netcoreapp1.1;net472;net461;net45;net40;net35;net20</CopyMeTestTargetFrameworks>
```

File: CodeJam.Main.Tests/TargetingTests.cs [hunk 3]
```diff
@@ -106,26 +108,115 @@ public class TargetingTests
 			net472
 			net48";
 
+		/// <summary> Full list of monikers for FW versions we do publish our packages </summary>
+		private const string _defaultFrameworkMoniker = "netcoreapp3.0";
+
+		/// <summary> Full list of monikers for FW versions we do publish our packages </summary>
+		private const string _packageFrameworkMonikers = @"
+			netcoreapp3.0
+			net5.0
```
[the remark this answers is not in the document]
Comment (ig-sinicyn) [reply CodeJam.Main.Tests/TargetingTests.cs:117]: VS does select the first moniker in list as a default target framework. We 're going to switch to the net core 3 ad our default target (i'ts too early to seriously invest in 5.0), so there it is.

File: CodeJam.Main.Tests/TargetingTests.cs [hunk 4]
```diff
@@ -106,26 +108,115 @@ public class TargetingTests
...
+			net45
+			net40
+			net35
+			net20";
+
+		/// <summary>Monikers for FW versions we do test our packages </summary>
+		// DONTTOUCH: the first one is a
+		private const string _testFrameworkMonikers = @"
+			netcoreapp3.0
+			net5.0
```
[the remark this answers is not in the document]
Comment (ig-sinicyn) [reply CodeJam.Main.Tests/TargetingTests.cs:135]: VS does select the first moniker in list as a default target framework, so netcoreapp3.0 should be first one.

File: CodeJam.Main.Tests/TargetingTests.cs [hunk 5]
```diff
@@ -60,6 +60,8 @@ public class TargetingTests
 			".NETCoreApp,Version=v2.0";
 #elif NETCOREAPP3_0
 			".NETCoreApp,Version=v3.0";
+#elif NET5_0
```
[the remark this answers is not in the document]
Comment (ig-sinicyn) [reply CodeJam.Main.Tests/TargetingTests.cs:65]: Thanks and fixed:)

File: Build/Props/CodeJam.Targeting.props (hunk 2 above)
Comment (ig-sinicyn) [reply Build/Props/CodeJam.Targeting.props:23]: VS does select the first moniker in list as a default target framework, so netcoreapp3.0 should be first one

File: Build/Props/CodeJam.Targeting.props (hunk 1 above)
Comment (ig-sinicyn) [reply Build/Props/CodeJam.Targeting.props:22]: VS does select the first moniker in list as a default target framework, so netcoreapp3.0 should be first one

File: CodeJam.Main.Tests/TargetingTests.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (ig-sinicyn) [reply CodeJam.Main.Tests/TargetingTests.cs:119]: > Why no .NET Core 3.1
We do not include netcoreapp 3.1 in list of target frameworks as it has have (almost) no changes in BCL from 3.0

[the remark this answers is not in the document]
Comment (NN---) [reply CodeJam.Main.Tests/TargetingTests.cs:119]: There should be a comment for this behavior then :)

[the remark this answers is not in the document]
Comment (ig-sinicyn) [reply CodeJam.Main.Tests/TargetingTests.cs:119]: Yep, will do

[the remark this answers is not in the document]
Comment (ig-sinicyn) [reply CodeJam.Main.Tests/TargetingTests.cs:119]: Done

**Review by NN--- — APPROVED** [review]

**Comment by NN---** [comment]
For some reason test expects 4.7.2 and not 4.8 .

```
Failed : CodeJam.TargetingTests.TestTargeting
  Expected string length 28 but was 26. Strings differ at index 25.
  Expected: ".NETFramework,Version=v4.7.2"
  But was:  ".NETFramework,Version=v4.8"
  ------------------------------------^
   at CodeJam.TargetingTests.TestTargeting() in C:\projects\codejam\CodeJam.Main.Tests\TargetingTests.cs:line 255
```

**Comment by NN---** [comment]
Fixed. Waiting for build.

**Comment by ig-sinicyn** [comment]
> Fixed. Waiting for build.

Thanks!

**Pull request merged**